notes-20200211.tex

```diff
@@ -0,0 +1,171 @@
+
+There is also an alternative formulation of $\idtype$-elim and $\idtype$-comp:
+
+% FIXME(remexre): The hack deepens...
+\begin{prooftree*}
+	\hypo{\istype{A}}
+	\hypo{\oftype{M}{A}}
+	\infer[no rule]2{\oftype{P}{\idtype{A}{M}{M}}}
+	\hypo{\istype{x{:}A,q{:}\idtype{A}{M}{x}}{C}}
+	\infer[no rule]1{\oftype{N}{C[x \rightarrow M, q \rightarrow \idrefl{M}]}}
+	\infer2[$\idtype$-elim$^{M=}$]{\oftype{\mathtt{J}^{M=}\brackets{x.q.C}\parens{N; P}}{C[x \rightarrow M, q \rightarrow \idrefl{M}]}}
+\end{prooftree*}
+\begin{prooftree*}
+	\hypo{\istype{A}}
+	\hypo{\oftype{M}{A}}
+	\infer[no rule]2{\oftype{P}{\idtype{A}{M}{M}}}
+	\hypo{\istype{x{:}A,q{:}\idtype{A}{M}{x}}{C}}
+	\infer[no rule]1{\oftype{N}{C[x \rightarrow M, q \rightarrow \idrefl{M}]}}
+	\infer2[$\idtype$-comp$^{M=}$]{\eqterm{\mathtt{J}^{M=}\brackets{x.q.C}\parens{N; \idrefl{M}}}{N}{C[x \rightarrow M, q \rightarrow \idrefl{M}]}}
+\end{prooftree*}
+
+This formulation is often easier to use in practice, but is equivalent to the one initially presented; see section 1.12.1 of \cite{hott-as:book} for details.
+
+\subsection{(Usual) Equality Types}
+
+Next, we introduce equality types $\mathtt{Eq}_A\parens{M; N}$. They are intended to be a direct internalization of judgmental equality by the following equality reflection rule:
+
+\begin{prooftree*}
+	\hypo{\oftype{\ctx}{p}{\mathtt{Eq}_A\parens{M; N}}}
+	\infer1{\eqterm{\ctx}{M}{N}{A}}
+\end{prooftree*}
+
+Note that by the substitutional nature of judgmental equality the converse holds as well:
+
+\begin{prooftree*}
+	\hypo{\eqterm{\ctx}{M}{N}{A}}
+	\infer1{\oftype{\ctx}{\idrefl{M}}{\mathtt{Eq}_A\parens{M; N}}}
+\end{prooftree*}
+
+In this sense, equality types and judgmental equality are equivalent. Formally, the rules of equality types consist of those of identification types and the equality reflection rule.
+
+\subsection{Reflection-Free Equality Types}
+A variant of equality types is reflection-free equality types \cite{sterling_et_al:LIPIcs:2019:10538}. They are intended to obtain the best parts of identification types and (ordinary) equality types. Reflection-free equality types are obtained from equality types by replacing the equality reflection rule by the following strict principle of uniqueness of identity proofs (UIP):
+
+\begin{prooftree*}
+	\hypo{\oftype{\ctx}{p}{\mathtt{Eq}_A\parens{M; N}}}
+	\hypo{\oftype{\ctx}{q}{\mathtt{Eq}_A\parens{M; N}}}
+	\infer2{\eqterm{\ctx}{p}{q}{\mathtt{Eq}_A\parens{M; N}}}
+\end{prooftree*}
+
+That is, UIP states that there is at most one term of a reflection-free equality type.
+Note that the principle is called \emph{strict} because the equality between terms of the same reflection-free equality type in the conclusion is judgmental. 
+
+Alternatively, we may employ Axiom K, which is equivalent to UIP in the sense that one of the two is derivable from the other; see \cite{hott-as:book} for the details. 
```

Comment:
It might be better to move this paragraph to the next session because the usual formulation of K/UIP is "weak" (not in terms of judgmental equality).